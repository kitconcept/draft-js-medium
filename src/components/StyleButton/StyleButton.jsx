/**
 * StyleButton component.
 * @module components/StyleButton
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from '../Button/Button';

/**
 * StyleButton component class.
 * @class StyleButton
 * @extends Component
 */
export default class StyleButton extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    icon: PropTypes.string.isRequired,
    style: PropTypes.string.isRequired,
    onToggle: PropTypes.func.isRequired,
  };

  /**
   * Construcor
   * @param {Object} props Properties.
   * @constructs
   */
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  /**
   * On click method.
   * @function onClick
   * @returns {undefined}
   */
  onClick() {
    this.props.onToggle(this.props.style);
  }

  /**
   * Render method.
   * @function render
   * @returns {string} Markup of the container.
   */
  render() {
    return <Button onClick={this.onClick} icon={this.props.icon} />;
  }
}
