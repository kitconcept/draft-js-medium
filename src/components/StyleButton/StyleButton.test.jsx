import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import StyleButton from './StyleButton';

jest.mock('../Button/Button', () =>
  jest.fn(({ onClick }) => <button onClick={onClick} />),
);

describe('StyleButton', () => {
  it('renders a style button component', () => {
    const component = renderer.create(
      <StyleButton icon="bold" style="BOLD" onToggle={() => {}} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('calls the onClick handler when button is pressed', () => {
    const handler = jest.fn();
    const wrapper = mount(
      <StyleButton icon="bold" style="BOLD" onToggle={handler} />,
    );
    wrapper.simulate('click');
    expect(handler).toBeCalled();
  });
});
