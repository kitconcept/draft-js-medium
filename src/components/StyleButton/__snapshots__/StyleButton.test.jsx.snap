// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleButton renders a style button component 1`] = `
<button
  onClick={[Function]}
/>
`;
